middle_end/flambda2/from_lambda/lambda_to_flambda_primitives.ml

```diff
@@ -381,11 +381,21 @@ let checked_alignment ~dbg ~primitive ~conditions : H.expr_primitive =
       dbg
     }
 
-let check_bound_tagged tagged_index bound : H.expr_primitive =
-  Binary
-    ( Int_comp (I.Naked_immediate, Yielding_bool (Lt Unsigned)),
-      untag_int tagged_index,
-      bound )
+let check_bound ~(index_kind : Lambda.array_index_kind) ~(bound_kind : I_or_f.t)
```

Comment:
I made this change. It required adding a conversion function to `Standard_int_or_float`, because as far as I can see there is no int-specific numeric conversion primitive, but I think it is worthwhile.